### PR TITLE
ENH/REF: GitHub Actions, setuptools-scm, pyproject.toml, ...

### LINKED ISSUE #24: Remove tests from this repo?
Or at least modify them.

> No idea... I think that the tests also can go away here... while reading them it looks like it is testing cookiecutter more than the product of this cookiecutter template... 
> I'm not sure if we need the tests here.

_Originally posted by @hhslepicka in https://github.com/pcdshub/cookiecutter-pcds-python/pull/23#issuecomment-639917675_

### LINKED ISSUE #35: Migrate to pyproject.toml for the generated project
`pyproject.toml` appears to be the future. `setup.py` is also not fun to write (though, in my opinion, it has its benefit in being actual lintable/inspectable Python code instead of a configuration file)

See https://github.com/pcdshub/pmpsdb_client/blob/master/pyproject.toml for example/inspiration

### LINKED ISSUE #36: Consider setuptools-scm instead of versioneer for future projects
[setuptools-scm](https://pypi.org/project/setuptools-scm/)

Working reference code:
* Installation: https://github.com/pcdshub/pmpsdb_client/blob/656eb6283d69bb40627cba19dc337b4b2074e328/pyproject.toml#L1-L3
* Building of `_version.py` https://github.com/pcdshub/pmpsdb_client/blob/656eb6283d69bb40627cba19dc337b4b2074e328/pyproject.toml#L24-L25

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Migrate from Travis CI to GitHub Actions
    * Remove doctr configuration
* Migrate from versioneer to setuptools-scm
   * Borrows experimental version handling by Zach in https://github.com/pcdshub/pmpsdb_client/blob/master/pmpsdb_client/version.py
* Migrate from setup.py/setup.cfg to pyproject.toml
* Move to year 2023
    * Support async entrypoint functions
* Fix documentation settings
   * Sphinx 6 now works 
      * Language was unset
      * jquery is no longer bundled (-> new requirement + extension)
   * docs-versions-menu works
* Add conda workflow support for setuptools-scm
    * I have not found a better way to do this - after wasting a morning - but am very open to input. 
    * The rest of the internet appears content with using pip to  test their package and conda-forge to handle the conda environment resolution/testing
* Streamline the issue template because we rarely fill in everything
* Miscellaneous cleanup

## Motivation and Context
* Migrating to GitHub actions, seemed like time to upgrade our standards overall
* Intend to use this cookiecutter output for further CI testing in https://github.com/pcdshub/pcds-ci-test-repo

Closes #24 
Closes #35 
Closes #36

### CODE REVIEW

File: {{ cookiecutter.folder_name }}/.github/workflows/standard.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches:
+      - master
+  release:
+    types:
+      - created
+
+jobs:
+  standard:
+    uses: klauer/pcds-ci-helpers/.github/workflows/python-standard.yml@enh_gha
```
Comment (klauer) [inline {{ cookiecutter.folder_name }}/.github/workflows/standard.yml:17]: TODO: change URL

Comment (klauer) [reply {{ cookiecutter.folder_name }}/.github/workflows/standard.yml:17]:
```suggestion
    uses: pcdshub/pcds-ci-helpers/.github/workflows/python-standard.yml
```

**Comment by klauer** [comment]
As a reminder from the text above, this was used to generate this repository successfully: https://github.com/pcdshub/pcds-ci-test-repo-python

File: {{ cookiecutter.folder_name }}/.github/workflows/standard.yml (hunk 1 above)
Comment (klauer) [reply {{ cookiecutter.folder_name }}/.github/workflows/standard.yml]: Closed in [adfc066](https://github.com/pcdshub/cookiecutter-pcds-python/pull/37/commits/adfc06659ee243670ff9ecd49fd7218482d6d6f9)

**Review by ZLLentz — APPROVED** [review]
I'm :+1: on all of this, looks great and I'll have to remember to use the template next time.

**Review by tangkong — APPROVED** [review]
Looks good to me.  👍   If we don't want to remove the `run_tests.py` instructions here I'm happy to do that in another PR

**Comment by klauer** [comment]
Good call - I'll make a quick follow-up PR @tangkong with some documentation changes so they don't get hidden here.

**Pull request merged**